Mathlib/RingTheory/MvPowerSeries/Basic.lean

```diff
@@ -0,0 +1,946 @@
+
+@[simp, norm_cast]
+theorem coe_monomial (n : σ →₀ ℕ) (a : R) :
+    (monomial n a : MvPowerSeries σ R) = MvPowerSeries.monomial R n a :=
+  MvPowerSeries.ext fun m => by
+    classical
+    rw [coeff_coe, coeff_monomial, MvPowerSeries.coeff_monomial]
+    split_ifs with h₁ h₂ <;> first |rfl|subst m; contradiction
+#align mv_polynomial.coe_monomial MvPolynomial.coe_monomial
+
+@[simp, norm_cast]
+theorem coe_zero : ((0 : MvPolynomial σ R) : MvPowerSeries σ R) = 0 :=
+  rfl
+#align mv_polynomial.coe_zero MvPolynomial.coe_zero
+
+@[simp, norm_cast]
+theorem coe_one : ((1 : MvPolynomial σ R) : MvPowerSeries σ R) = 1 :=
+    coe_monomial _ _
+#align mv_polynomial.coe_one MvPolynomial.coe_one
+
+@[simp, norm_cast]
+theorem coe_add : ((φ + ψ : MvPolynomial σ R) : MvPowerSeries σ R) = φ + ψ :=
+  rfl
+#align mv_polynomial.coe_add MvPolynomial.coe_add
+
+@[simp, norm_cast]
+theorem coe_mul : ((φ * ψ : MvPolynomial σ R) : MvPowerSeries σ R) = φ * ψ :=
+  MvPowerSeries.ext fun n => by
+    classical
+    simp only [coeff_coe, MvPowerSeries.coeff_mul, coeff_mul]
+#align mv_polynomial.coe_mul MvPolynomial.coe_mul
+
+@[simp, norm_cast]
+theorem coe_C (a : R) : ((C a : MvPolynomial σ R) : MvPowerSeries σ R) = MvPowerSeries.C σ R a :=
+  coe_monomial _ _
+set_option linter.uppercaseLean3 false in
+#align mv_polynomial.coe_C MvPolynomial.coe_C
+
+set_option linter.deprecated false in
+@[simp, norm_cast]
+theorem coe_bit0 :
+    ((bit0 φ : MvPolynomial σ R) : MvPowerSeries σ R) = bit0 (φ : MvPowerSeries σ R) :=
+  coe_add _ _
+#align mv_polynomial.coe_bit0 MvPolynomial.coe_bit0
+
+set_option linter.deprecated false in
+@[simp, norm_cast]
+theorem coe_bit1 :
+    ((bit1 φ : MvPolynomial σ R) : MvPowerSeries σ R) = bit1 (φ : MvPowerSeries σ R) := by
+  rw [bit1, bit1, coe_add, coe_one, coe_bit0]
+#align mv_polynomial.coe_bit1 MvPolynomial.coe_bit1
+
+@[simp, norm_cast]
+theorem coe_X (s : σ) : ((X s : MvPolynomial σ R) : MvPowerSeries σ R) = MvPowerSeries.X s :=
+  coe_monomial _ _
+set_option linter.uppercaseLean3 false in
+#align mv_polynomial.coe_X MvPolynomial.coe_X
+
+variable (σ R)
+
+theorem coe_injective : Function.Injective (Coe.coe : MvPolynomial σ R → MvPowerSeries σ R) :=
+    fun x y h => by
+  ext
+  simp_rw [← coeff_coe]
+  congr
+#align mv_polynomial.coe_injective MvPolynomial.coe_injective
+
+variable {σ R φ ψ}
+
+@[simp, norm_cast]
+theorem coe_inj : (φ : MvPowerSeries σ R) = ψ ↔ φ = ψ :=
+  (coe_injective σ R).eq_iff
+#align mv_polynomial.coe_inj MvPolynomial.coe_inj
+
+@[simp]
+theorem coe_eq_zero_iff : (φ : MvPowerSeries σ R) = 0 ↔ φ = 0 := by rw [← coe_zero, coe_inj]
+#align mv_polynomial.coe_eq_zero_iff MvPolynomial.coe_eq_zero_iff
+
+@[simp]
+theorem coe_eq_one_iff : (φ : MvPowerSeries σ R) = 1 ↔ φ = 1 := by rw [← coe_one, coe_inj]
+#align mv_polynomial.coe_eq_one_iff MvPolynomial.coe_eq_one_iff
+
+/-- The coercion from multivariable polynomials to multivariable power series
+as a ring homomorphism.
+-/
```

Comment:
```suggestion
/-- The coercion from multivariable polynomials to multivarite power series
as a ring homomorphism.
-/
```
Here and below, multivariable should become multivariate in accordance with the beginning of the file.